pkg/server/io/container_io.go

```diff
@@ -86,13 +88,29 @@ func NewContainerIO(id string, opts ...ContainerIOOpts) (_ *ContainerIO, err err
 	if c.fifos == nil {
 		return nil, errors.New("fifos are not set")
 	}
-	// Create actual fifos.
-	stdio, closer, err := newStdioPipes(c.fifos)
+
+	u, err := url.Parse(c.fifos.Stdout)
+
 	if err != nil {
 		return nil, err
 	}
-	c.stdioPipes = stdio
-	c.closer = closer
+
+	if u.Scheme == "" {
+		u.Scheme = "fifo"
+	}
+
+	switch u.Scheme {
```

Comment:
Do we expect that there may be cases other than fifo or binary here in the future?

Reply:
no, I don't think so.